maestro-ios/src/main/java/ios/simctl/SimctlIOSDevice.kt

```diff
@@ -123,6 +121,11 @@ class SimctlIOSDevice(
         TODO("Not yet implemented")
     }
 
+    override fun setPermissions(id: String, permissions: Map<String, String>) {
+        Simctl.setPermissions(deviceId, id, permissions)
+        xcTestDevice.restartXCTestRunnerService()
```

Comment:
Because it is killed

Reply:
Ohh interesting, yes I remember in cloud we needed to restart idb as well can you check if idb operations work as well after this ?

Reply:
Good point, I'll test this

Reply:
Tested, idb works fine after `setPermissions`

Reply:
Not every flow has a launchApp command, so that can only be done by first inspecting if the flow starts with a launchApp command. Which is not easy because the first command could be a runFlow or a runScript command, etc.

Reply:
This is future improvement but I see it as follows:

1. Whenever `setPermissions` is called in test initially with `launchApp` we run the xctest runner after that
2. Subsequently, after that, all the `setPermissions` calls will not install/run xctest runner again

This will avoid installing xctest runner multiple times, right now its done in open as well as again everytime setPermission is called as part of launchApp.

Reply:
We should definitely look into speeding up the startup steps we do here. I've measured, this change introduces ~1.5 extra time on a flow start with launchApp with clearState. I think that is acceptable. Speed improvements are always welcome..